src/tests/krate/yanking.rs

```diff
@@ -91,7 +91,7 @@ fn test_yank_rate_limiting() {
     check_yanked(&app, true);
 
     // Let the rate limit refill.
-    std::thread::sleep(Duration::from_millis(500));
```

Comment:
oh, good question! I dug into it and found that actually it wasn't caused by this short sleep. 

I tried to limit the resources of the database. Then I found if your database is too slow, then there is a change it just refills the rate automatically and gives you the valid token.

So I think the right solution for this issue is that we can increase the rate, and then we can tolerate the database being a little bit slow.

Reply:
I limited the database with docker-compose:
```yaml
services:
  postgres:
    image: postgres:13
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: password
    ports:
      - 127.0.0.1:5432:5432
    volumes:
      - ./docker/docker-entrypoint-initdb.d:/docker-entrypoint-initdb.d
      - postgres-data:/var/lib/postgresql/data
    deploy:
      resources:
        limits:
          cpus: '0.01'
          memory: 500M
        reservations:
          cpus: '0.01'
          memory: 200M
```

Reply:
tbh I'm not a huge fan of increasing the timeouts in this brittle test. if someone has an even slower system for some reason then the test would still fail.

I'm wondering if we could do something closer to what we did in https://github.com/rust-lang/crates.io/pull/7105 to resolve this problem 🤔 

also, for anyone else reading along, it looks like `assert_rate_limited()` was implemented with `assert_eq(expected, actual)` instead of the `assert_eq(actual, expected)` ordering we use everywhere else. this can be a bit confusing since Rust only outputs `left` and `right` if an assertion fails.

Reply:
Updated. I split it into two tests. Thanks!

I also have updated the assert_rate_limited to use the correct order.